### DISCUSSION TITLE
Re-usable media/com_content/js/admin-articles-default-batch-footer.js

### DISCUSSION THREAD

**Discussion by joomdonation** [joomla/joomla-cms#34980] [2. Cold Features]
The above mentioned script is written to use for com_content only. Can it somehow, be used for other components such as com_banners, com_contacts....? Currently, in these components, we still use inline JS. Ping @dgrammatiko @Fedik for help.

**Comment by Fedik** [comment]
yes
```php
/** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
$wa = $this->document->getWebAssetManager();
$wa->getRegistry()->addExtensionRegistryFile('com_content'); // <- Add com_content assets to registry, manually
$wa->useScript('com_content.admin-articles-batch');
```

**Reply by joomdonation** [reply]
I know. The problem is that we hardcode the task to com_content https://github.com/joomla/joomla-cms/blob/4.0-dev/build/media_source/com_content/js/admin-articles-default-batch-footer.es6.js#L18

And just a question: Is it safe to use assets from a different extension in another extension?

**Reply by Fedik** [reply]
> Is it safe to use assets from a different extension in another extension?

As long as you wear a helmet ;) 
We have use some stuff from categories and lang associations, in this way.

> The problem is that we hardcode the task

I see now, it will not work for others, as is.
In theory can read the task from batch button (as it done in set up), 

Or well, just remove that `if (form && task === 'article.batch')`, whole `submitTask()` cannot be called externally, but only if someone click `#batch-submit-button-id`
